Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp

```diff
@@ -340,13 +342,19 @@ void AuthenticatorCoordinator::discoverFromExternalSource(const Document& docume
         });
     }
 
-    auto callback = [weakThis = WeakPtr { *this }, promise = WTFMove(promise), abortSignal = WTFMove(requestOptions.signal)] (AuthenticatorResponseData&& data, AuthenticatorAttachment attachment, ExceptionData&& exception) mutable {
+    auto callback = [weakThis = WeakPtr { *this }, promise = WTFMove(promise), abortSignal = WTFMove(requestOptions.signal), weakDocument = WeakPtr { document }] (AuthenticatorResponseData&& data, AuthenticatorAttachment attachment, ExceptionData&& exception) mutable {
         if (abortSignal && abortSignal->aborted()) {
             promise.reject(Exception { ExceptionCode::AbortError, "Aborted by AbortSignal."_s });
             return;
         }
 
         if (auto response = AuthenticatorResponse::tryCreate(WTFMove(data), attachment)) {
+            if (RefPtr document = weakDocument.get()) {
+                RefPtr page = document->page();
+                if (!page)
+                    return;
+                page->setLastAuthentication(LoginStatus::AuthenticationType::WebAuthn);
+            }
```

Comment:
You can just capture page instead of document in this lambda: `weakPage = WeakPtr { document.page() }`.

Then this could be changed to:
```suggestion
            if (RefPtr page = weakPage.get())
                page->setLastAuthentication(LoginStatus::AuthenticationType::WebAuthn);
```